src/language-js/needs-parens.js

```diff
@@ -241,6 +241,19 @@ function needsParens(path, options) {
         return true;
       }
       break;
+
+    // A user typing `!foo instanceof Bar` probably intended
+    // `!(foo instanceof Bar)`, so format to `(!foo) instance Bar` to what is
+    // really happening
+    case "BinaryExpression":
+      if (
+        key === "left" &&
+        (parent.operator === "in" || parent.operator === "instanceof") &&
+        node.type === "UnaryExpression"
```

Comment:
If we are not going to check operator, please add tests for all the operators. 

https://github.com/estree/estree/blob/master/es5.md#unaryoperator

Reply:
Done!